src/test/java/io/cdap/plugin/salesforce/SalesforceSchemaUtilTest.java

```diff
@@ -166,6 +168,40 @@ public void testGetSchemaWithFields() {
     new org.apache.avro.Schema.Parser().parse(actualSchema.toString());
   }
 
+  @Test
+  public void getSchemaWithNullsAllowedInNonNullCustomFields() {
+    String objectName = "CustomTable";
+
+    List<SObjectDescriptor.FieldDescriptor> fieldDescriptors = Stream
+            .of("Name", "Value", "CreatedDate")
+            .map(name -> getFieldWithType(name, FieldType.anyType, false))
+            .map(SObjectDescriptor.FieldDescriptor::new)
+            .collect(Collectors.toList());
+    fieldDescriptors.add(new SObjectDescriptor.FieldDescriptor(
+            Collections.singletonList("Name"), null, SalesforceFunctionType.NONE));
+    fieldDescriptors.add(new SObjectDescriptor.FieldDescriptor(
+            Collections.singletonList("Value"), null, SalesforceFunctionType.NONE));
+    fieldDescriptors.add(new SObjectDescriptor.FieldDescriptor(
+            Collections.singletonList("CreatedDate"), null, SalesforceFunctionType.NONE));
+    SObjectDescriptor sObjectDescriptor = new SObjectDescriptor(objectName, fieldDescriptors, ImmutableList.of());
+
+    Map<String, Field> objectFields = new LinkedHashMap<>();
+    objectFields.put("Name", getCustomFieldWithType("Name", FieldType.string, false));
+    objectFields.put("Value", getCustomFieldWithType("Value", FieldType.currency, false));
+    objectFields.put("CreatedDate", getCustomFieldWithType("CreatedDate", FieldType.date, false));
+    SObjectsDescribeResult describeResult = SObjectsDescribeResult.of(ImmutableMap.of(objectName, objectFields));
+
+    Schema expectedSchema = Schema.recordOf("output",
+            Schema.Field.of("Name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+            Schema.Field.of("Value", Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+            Schema.Field.of("CreatedDate", Schema.nullableOf(Schema.of(Schema.LogicalType.DATE)))
+    );
+
+    Schema actualSchema = SalesforceSchemaUtil.getSchemaWithFields(sObjectDescriptor, describeResult, true);
```

Comment:
Can you add another validation in this function with 'false' ?

Reply:
Done